read-fonts/src/table_ref.rs

```diff
@@ -31,3 +33,28 @@ impl<'a, T> TableRef<'a, T> {
 impl<U, T: Format<U>> Format<U> for TableRef<'_, T> {
     const FORMAT: U = T::FORMAT;
 }
+
+/// Combination of a tag and a child table.
+///
+/// This is used in cases where a data structure has an array of records
```

Comment:
instead of being overly general I'd just say it's used in the script & feature lists? Do we anticipate any other cases?

Reply:
There are three cases here. I did think this pattern was more general but I'm not coming up with any additional cases. 

Going to move the type to the `layout` module and I'll narrow the docs. We can change in the future if other cases come up.